src/main/java/org/jenkinsci/plugins/configfiles/buildwrapper/ManagedFile.java

```diff
@@ -131,18 +133,23 @@ public ListBoxModel doFillFileIdItems(@AncestorInPath ItemGroup context, @Ancest
         public HttpResponse doCheckFileId(StaplerRequest req, @AncestorInPath Item context, @QueryParameter String fileId) {
             // You should have permission to configure your project in order to check whether the selected file id is
             // allowed to you
-            if (context != null) {
-                context.checkPermission(Item.CONFIGURE);
-            }
-            
-            final Config config = ConfigFiles.getByIdOrNull(context, fileId);
-            if (config != null) {
-                return ConfigFileDetailLinkDescription.getDescription(req, context, fileId);
+            Permission permission = context == null ? Item.CREATE : Item.CONFIGURE;
+            AccessControlled ac = context == null ? Jenkins.get() : context;
+            ac.checkPermission(permission);
```

Comment:
maybe it is just me but the multiple ternary operators are hard to read here.

```suggestion
            if (context!= null) {
                context.checkPermission(Item.CONFIGURE);
            } else {
                Jenkins.get().checkPermission(Item.CREATE);
            }
```